frontend/src/features/qa/page-list/ui/page-details.ts

```diff
@@ -0,0 +1,108 @@
+import { msg } from "@lit/localize";
+import { html, nothing } from "lit";
+
+import { crawlCounts } from "../helpers/crawlCounts";
+import { iconFor } from "../helpers/iconFor";
+import {
+  severityFromMatch,
+  severityFromResourceCounts,
+} from "../helpers/severity";
+
+import type { ArchivedItemQAPage } from "@/types/qa";
+import { tw } from "@/utils/tailwind";
+
+export function formatPercentage(n: number) {
+  if (Number.isNaN(n)) {
+    return "n/a";
+  }
+  return (n * 100).toFixed(2).replace(/[.,]00$/, "");
+}
+
+export const pageDetails = (page: ArchivedItemQAPage) =>
+  html`<ul class="leading-4">
+      <li class="my-3 flex">
+        ${iconFor(
+          severityFromMatch(page.qa.screenshotMatch),
+          tw`mr-2 flex-none`,
+        )}
+        <span class="inline-block">
+          ${page.qa.screenshotMatch != null
```

Comment:
Is `screenshotMatch` ever going to be null? I remember it could be at one point, did we perform a migration to ensure it never is? If not, let's update the type here.

Reply:
Yes, we should assume that this may be optional, as all of the QA data may not be available for whatever reason (screenshot, text, resources)